src/Framework/PaymentGateways/Contracts/PaymentGateway.php

```diff
@@ -2,11 +2,46 @@
 
 namespace Give\Framework\PaymentGateways\Contracts;
 
+use Give\Framework\Exceptions\Primitives\InvalidArgumentException;
 use Give\Framework\LegacyPaymentGateways\Contracts\LegacyPaymentGatewayInterface;
 
 /**
  * @unreleased
  */
 abstract class PaymentGateway implements PaymentGatewayInterface, LegacyPaymentGatewayInterface {
+	/**
+	 * @var SubscriptionModuleInterface $subscriptionModule
+	 */
+	public $subscriptionModule;
+
+	/**
+	 * @inheritDoc
+	 */
+	public function mountSubscriptionModule( $subscriptionModuleClass ) {
+		if ( ! class_implements( $subscriptionModuleClass, SubscriptionModuleInterface::class ) ) {
+			throw new InvalidArgumentException( sprintf(
+				'%1$s must implement %2$s',
+				$subscriptionModuleClass,
+				SubscriptionModuleInterface::class
+			) );
+		}
+
+		$this->subscriptionModule = give( $subscriptionModuleClass );
+	}
```

Comment:
@jonwaldstein I'm thinking of changing up this mounting process slightly:
1. Remove this method
2. Make the `SubscriptionModuleInterface` a nullable parameter of the `PaymentGateway` constructor
3. In the `singleton` function, instead of passing this class, just grab the interface:
    ```php
    $subscriptionModule = apply_filters("give_gateway_{$gatewayId}_subscription_module", null);
    return $container->make($gatewayClass, [$subscriptionModule ? $container->make($subscriptionModule) : null);
    ```

Reply:
I like this idea. It's less code and the filter has a very specific scope. Updated 9a13693a

Side note about my issue with using the `$container->make` inside the singleton callback: 

I was able to use the `$container-make()` _inside_ the class params when instantiating - but not outside that / returning or else it will crash my site.